Lombiq.Hosting.MediaTheme.Bridge/Services/MediaThemeManager.cs

```diff
@@ -77,11 +79,20 @@ public async Task UpdateBaseThemeAsync(string baseThemeId)
 
     public async Task<MediaTemplate> GetMediaTemplateByShapeTypeAsync(string shapeType)
     {
-        var templatePath = _mediaFileStore.Combine(
-            Paths.MediaThemeRootFolder,
-            Paths.MediaThemeTemplatesFolder,
-            shapeType + ".liquid");
-        if (!await _mediaFileStore.FileExistsAsync(templatePath)) return null;
+        var templatePath = GetPath(shapeType + LiquidExtension);
+
+        // Check if the file exists with a different casing, before giving up.
+        if (!await _mediaFileStore.FileExistsAsync(templatePath))
+        {
+            var basePath = GetPath();
+            var file = await _mediaFileStore
+                .GetDirectoryContentAsync(basePath)
+                .Where(file => file.Name.EqualsOrdinalIgnoreCase(shapeType + LiquidExtension))
+                .FirstOrDefaultAsync();
+
+            if (file == null) return null;
+            templatePath = GetPath(file.Name);
+        }
```

Comment:
Should this be removed if https://github.com/OrchardCMS/OrchardCore/issues/16585 is fixed? If yes, comment it here.

Reply:
I'm not sure yet, it depends how that issue gets resolved. I've added a comment accordingly.